pkg/app/pipedv1/plugin/kubernetes/deployment/server.go

```diff
@@ -175,3 +184,72 @@ func (a *DeploymentService) loadManifests(ctx context.Context, deploy *model.Dep
 
 	return manifests, nil
 }
+
+func (a *DeploymentService) ExecuteStage(ctx context.Context, request *deployment.ExecuteStageRequest) (*deployment.ExecuteStageResponse, error) {
+	switch request.GetInput().GetStage().GetName() {
+	case StageK8sSync.String():
+		return a.executeK8sSyncStage(ctx, request.GetInput())
+	case StageK8sRollback.String():
+		return a.executeK8sRollbackStage(ctx, request.GetInput())
+	default:
+		return nil, status.Error(codes.InvalidArgument, "unimplemented or unsupported stage")
+	}
+}
+
+func (a *DeploymentService) executeK8sSyncStage(ctx context.Context, input *deployment.ExecutePluginInput) (*deployment.ExecuteStageResponse, error) {
+	lp := a.LogPersister.StageLogPersister(input.GetDeployment().GetId(), input.GetStage().GetId())
```

Comment:
Be sure to call `lp.Complete(timeout)` once we finish this stage execution so that the log persisted bulk can be clean up properly
ref: https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/controller/scheduler.go#L509 (for how to close in case of pipedv0)

Reply:
@khanhtc1202
Thank you for letting me know that point.
It's okay to call lp.Complete on the plugin side. but I think it's a control-flow operation, so it's nice to be done on the piped side. WDYT?

Reply:
The current implementation of log persister on the piped side is just a proxy to push logs sent by plugins to the control plane (ref: https://github.com/pipe-cd/pipecd/blob/master/pkg/app/pipedv1/cmd/piped/service/service.proto#L30-L33), it has no idea with the log persister object, so I can not image how the piped can send a signal to complete a log persister object which is created in the plugin. WDYT? 👀 
Also, since a logpersister (lp) object is a stage-specific object, it could be easy to "close" it when the stage execution is completed. @Warashi

Reply:
I read the codes and realized that I misunderstood the log persister. As you said, it's challenging to do on the piped side. Thank you.

Reply:
@khanhtc1202 
fixed on this commit.
https://github.com/pipe-cd/pipecd/pull/5371/commits/3233eaffb4352672e4c25c642c2c4025a1adb0c0